lib/src/code_generator/func_type.dart

```diff
@@ -81,6 +81,18 @@ class FunctionType extends Type {
     return sb.toString();
   }
 
+  @override
+  bool get sameFfiDartAndCType =>
```

Comment:
These two method implementations being equal makes me think we should have `ffiDartType() == cType()` on the call sites instead.

(Some subclasses return a bool immediately now, so it's slightly less performant, but I believe doing equality is cleaner from a code-style point of view.)